src/Reflection/ReflectionXHPChildrenDeclaration.hack

```diff
@@ -52,18 +51,17 @@ class ReflectionXHPChildrenDeclaration {
   <<__Memoize>>
   public function getExpression(): ReflectionXHPChildrenExpression {
     try {
-      $data = TypeSpec\varray(TypeSpec\mixed())
-        ->assertType($this->data);
-      return new ReflectionXHPChildrenExpression($this->context, $data);
-    } catch (IncorrectTypeException $_) {
-      // handled below
+      return new ReflectionXHPChildrenExpression(
+        $this->context,
+        $this->data as KeyedContainer<_, _>,
```

Comment:
```suggestion
        $this->data as Container<_>,
```
otherwise, this will throw for a list. no?

Reply:
The list-likes (varray, vec, ConstVector and Co) are all `KeyedContainer<int, T>`. This expression will not throw.

Reply:
traversables?

Reply:
The ReflectionXHPChildrenExpression keys into this argument in a vec-ish way, so Traversable would not work.
The previous version of the code was equivalent to `TypeAssert\matches<varray<mixed>>($this->data)`.
`$this->data as varray<_>` would be a fine translation of that, but this is not allowed, since `as varray` is illegal.
I use `as KeyedContainer<_, _>`, as this still permits a varray.

```diff
-      $data = TypeSpec\varray(TypeSpec\mixed())
-        ->assertType($this->data);
-      return new ReflectionXHPChildrenExpression($this->context, $data);
-    } catch (IncorrectTypeException $_) {
+      return new ReflectionXHPChildrenExpression(
+        $this->context,
+        $this->data as KeyedContainer<_, _>,
+      );
+    } catch (\TypeAssertionException $_) {
```